generators/app/templates/src/_plugin-advanced.js

```diff
@@ -0,0 +1,47 @@
+import videojs from 'video.js';
+import {version as VERSION} from '../package.json';
+
+const Plugin = videojs.getPlugin('plugin');
+
+// Default options for the plugin.
+const defaults = {};
+
+/**
+ * An advanced Video.js plugin.
+ */
+class <%= pluginClassName %> extends Plugin {
+
+  /**
+   * Create a <%= pluginClassName %> plugin instance.
+   *
+   * @param  {Player} player
+   *         A Video.js Player instance.
+   *
+   * @param  {Object} [options]
+   *         An optional options object.
+   *
+   *         While not a core part of the Video.js plugin architecture, a
+   *         second argument of options is a convenient way to accept inputs
+   *         from your plugin's caller.
+   */
+  constructor(player, options) {
+    super(player);
+
+    this.options = videojs.mergeOptions(defaults, options);
```

Comment:
Maybe add a `this.player = player` here too?

Reply:
That happens implicitly in `super()`. Maybe I'll add a comment indicating that `this.player` is available.

Reply:
We should link to the docs and blog posts and such.